### PR TITLE
chore(analytics): add release and session to sentry

### PR DESCRIPTION
**Pull request by kevinslin**
chore(analytics): add release and session to sentry

- this adds `sessionId` as a tag in sentry 
- this also adds `release` to sentry so we can start tracking [release health](https://docs.sentry.io/product/releases/health/)
- cleanup
    - converts app targets like "vscode" and "cli" into an enum
    - since we bumped the min version, updated the following
        - remove check for old version of vscode when checking for workspace trust
        - remove `VSCodeEvents.UserOnOldVSCodeVerUnblocked` event 
    - remove commented code

Next Items
- [ ] SOP on how to make use of release health data
- [ ] track `sessionId` in server (this is a bigger task and out of scope for this pr)
- [ ] add sentry to cli
***
# Dendron Extended PR Checklist

### Basics

- [x] code should follow [Code Conventions](https://wiki.dendron.so/notes/773e0b5a-510f-4c21-acf4-2d1ab3ed741e.html)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](https://wiki.dendron.so/notes/pMS27sHxbWeKMoPRrWEzs.html).
- [x] sticking to existing conventions instead of creating new ones 
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended

- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [x] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed 
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics

- [x] if you are adding analytics related changes, make sure you [Document](https://wiki.dendron.so/notes/8ThPaB9iXXm2Szk3C9kFt.html) changes in airtable

### Extended

- [x] can we track the performance of this change to know if it is _successful_? 
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

## 

### Basics

- [~] [Write Tests](https://wiki.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html) 
- [x] [Confirm existing tests pass](https://wiki.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html)
- [x] [Confirm manual testing](https://wiki.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html) 
- [~] Common cases tested
- [~] 1-2 Edge cases tested
- [~] If your tests changes an existing snapshot, snapshots have been [updated](https://wiki.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html)

### Extended

NA

### Basics

- [~] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR

## 

### Basics

- [~] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](https://wiki.dendron.so/notes/3489b652-cd0e-4ac8-a734-08094dc043eb.html) or [Packages](https://wiki.dendron.so/notes/32cdd4aa-d9f6-4582-8d0c-07f64a00299b.html)

## 

### Extended

- [~]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](https://wiki.dendron.so/notes/iZ8vpfY0n1E3Nq3IC1Y7X.html)
  - eg. [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/common-server/src/errorReporting.ts [hunk 1]
```diff
@@ -39,10 +40,29 @@ export function isBadErrorThatShouldBeSampled(
...
+  environment: Stage;
+  sessionId?: number;
+  release: string;
+}): void {
   const dsn =
     "https://bc206b31a30a4595a2efb31e8cc0c04e@o949501.ingest.sentry.io/5898219";
 
+  const initialScope: CaptureContext = {};
+  if (sessionId) {
+    initialScope.tags = { sessionId };
```
Comment (SeriousBug) [inline packages/common-server/src/errorReporting.ts:63]: For future reference: if I wanted to add more properties to the user in Sentry (for example whether they have self contained vaults enabled or not), would this be where to add it?

**Review by SeriousBug — APPROVED** [review]
Looks good to me, just curious about one thing.

File: packages/common-server/src/errorReporting.ts (hunk 1 above)
Comment (kevinslin) [reply packages/common-server/src/errorReporting.ts:63]: yep, just wrote docs as part of an async
